mlflow/shap.py

```diff
@@ -179,3 +257,323 @@ def predict_function(X) -> pred:
+            for pip_dependency in dependency["pip"]:
+                if pip_dependency != "mlflow":
+                    merged_pip_deps.add(pip_dependency)
+        else:
+            if dependency.split("=")[0] != "python" and dependency.split("=")[0] != "pip":
+                merged_conda_deps.add(dependency)
+
+    merged_conda_deps = list(merged_conda_deps)
+    merged_pip_deps = list(merged_pip_deps)
+
+    return _mlflow_conda_env(
+        additional_conda_deps=merged_conda_deps,
+        additional_pip_deps=merged_pip_deps,
+        additional_conda_channels=merged_conda_channels,
+    )
+
+
+@experimental
+def load_explainer(model_uri):
+    """
+    Load a SHAP explainer from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model, for example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                      - ``models:/<model_name>/<model_version>``
+                      - ``models:/<model_name>/<stage>``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+                      artifact-locations>`_.
+
+    :return: A SHAP explainer.
+    """
+
+    explainer_path = _download_artifact_from_uri(artifact_uri=model_uri)
+    flavor_conf = _get_flavor_configuration(model_path=explainer_path, flavor_name=FLAVOR_NAME)
+    explainer_artifacts_path = os.path.join(explainer_path, flavor_conf["serialized_explainer"])
+    underlying_model_flavor = flavor_conf["underlying_model_flavor"]
+    model = None
+
+    if underlying_model_flavor != _UNKNOWN_MODEL_FLAVOR:
+        underlying_model_path = os.path.join(explainer_path, _UNDERLYING_MODEL_SUBPATH)
+        if underlying_model_flavor == "sklearn":
+            model = mlflow.sklearn._load_pyfunc(underlying_model_path).predict
+        elif underlying_model_flavor == "pytorch":
+            model = mlflow.pytorch._load_model(os.path.join(underlying_model_path, "data"))
+
+    return _load_explainer(explainer_file=explainer_artifacts_path, model=model)
+
+
+@experimental
+def _load_explainer(explainer_file, model=None):
+    """
+    Load a SHAP explainer saved as an MLflow artifact on the local file system.
+
+    :param explainer_file: Local filesystem path to the MLflow Model saved with the ``shap`` flavor
+    :param model: model to override underlying explainer model.
+    """
+    import pickle
+    import shap
+
+    def inject_model_loader(in_file):
+        pickle.load(in_file)  # No-Op to move file pointer forward
+        return model
+
+    with open(explainer_file, "rb") as explainer:
+        model_loader = None
+        if model is not None:
+            model_loader = inject_model_loader
```

Comment:
```suggestion
        model_loader = None if model is None else inject_model_loader
```

Can we use a ternary operator here (which I think is a bit easier to understand)?

Reply:
Changed as suggested